src/pymodaq_plugins_arduino/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Analog.py

```diff
@@ -0,0 +1,208 @@
+    ##
+
+    def ini_detector(self, controller=None):
+        """Detector communication initialization
+
+        Parameters
+        ----------
+        controller: (object)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one actuator/detector by controller
+            (Master case)
+
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+
+        self.ini_detector_init(slave_controller=controller)
+
+        if self.is_master:
+            self.controller = Arduino(com_port=self.settings['com_port'])  # instantiate you driver with whatever arguments are needed
+
+
+        # TODO for your custom plugin (optional) initialize viewers panel with the future type of data
+        self.dte_signal_temp.emit(DataToExport(name='AnalogInput',
+                                               data=[DataFromPlugins(name='Mock1',
+                                                                     data=[np.array([0]), np.array([0]),
+                                                                           np.array([0]), np.array([0]),
+                                                                           np.array([0]), np.array([0])],
+                                                                     dim='Data0D',
+                                                                     labels=['AI0', 'AI1', 'AI2', 'AI3', 'AI4',
+                                                                             'AI5'])]))
+        #self.controller.analog_pin_values_input
+        info = "Analog ready"
+        initialized = True
+        return info, initialized
+
+    def close(self):
+        """Terminate the communication protocol"""
+        if self.is_master:
+            self.controller.shutdown()
+
+    def grab_data(self, Naverage=1, **kwargs):
+        """Start a grab from the detector
+
+        Parameters
+        ----------
+        Naverage: int
+            Number of hardware averaging (if hardware averaging is possible, self.hardware_averaging should be set to
+            True in class preamble and you should code this implementation)
+        kwargs: dict
+            others optionals arguments
+        """
+        #self.settings.child('')
+        data_tot=[]
+
+        for param in self.settings.children():
+            if 'AI' in param.name():
+                if param['ch']:
+                    self.controller.set_analog_input(int(param.name()[2:3]))
+                    data_tot.append(np.array([self.controller.analog_pin_values_input[int(param.name()[2:3])]]))
+                else:
+                    data_tot.append(np.array([0]))
+
+        self.dte_signal.emit(DataToExport(name='Analog Input',
+                                          data=[DataFromPlugins(name='AI', data=data_tot,
+                                                                dim='Data0D', labels=['AI0', 'AI1', 'AI2', 'AI3', 'AI4',
```

Comment:
if I'm correct, you won't have alsways the 5 AI channels activated therefore the labels won't always be as you wrote. You could construct this labels list in the same way you construct data_tot, ie appending the label string (param name or other) if param['ch'] is true